app/src/main/java/com/amaze/filemanager/database/UtilsHandler.java

```diff
@@ -126,6 +128,53 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
         SFTP
     }
 
+    public void saveToDatabase(OperationData operationData) {
+        AppConfig.runInBackground(() -> {
+            switch (operationData.type) {
+                case HIDDEN:
+                case HISTORY:
+                case LIST:
+                case GRID:
+                    setPath(operationData.type, operationData.path);
+                    break;
+                case BOOKMARKS:
+                case SMB:
+                    setPath(operationData.type, operationData.name, operationData.path);
+                    break;
+                case SFTP:
+                    addSsh(operationData.name, operationData.path, operationData.hostKey,
```

Comment:
`addSsh()` with private key included would crash app on creating new SCP/SFTP connection because the async save-to-database operation didn't give enough time for LoadFilesListTask to login with key authentication ([SftpConnectDialog line 331](https://github.com/EmmanuelMess/AmazeFileManager/blob/a6fb7df4ca2af66aa8a55d8f1783354d432d243a/app/src/main/java/com/amaze/filemanager/ui/dialogs/SftpConnectDialog.java#L331)).

Verified on Oneplus One running Slim7 (7.1.2).

Must confess it was a flaw that I didn't resolve when I worked on #863. I'll see if I can work around this... most likely with an optional callback?

Reply:
>  because the async save-to-database operation didn't give enough time for LoadFilesListTask to login with key authentication

This is by design? If so it should be fixed by you ASAP.